src/shared/vpick.c

```diff
@@ -185,11 +185,9 @@ static int pin_choice(
         if (!result.path)
                 return log_oom_debug();
 
-        if (filter->version) {
-                result.version = strdup(filter->version);
-                if (!result.version)
-                        return log_oom_debug();
-        }
+        r = strdup_to(&result.version, filter->version);
+        if (r < 0)
+                return r;
```

Comment:
`log_oom_debug()` ?

Reply:
Hmm, so I'm hesitant about this. Generally in library-like codes it's better to propagate the original error, but that means log_oom_debug can't be used...

Reply:
Yeah, but it only returns -ENOMEM on error... There are so many places we use log_oom() or log_oom_debug() such simple allocation failure with `r == -ENOMEM`.

Reply:
Hmm, but this is library-like code? We only do < 0 -> log_oom() when the function returns NULL on failure or in user-facing executables?

Reply:
log_oom_debug() is not the same as log_oom()

Reply:
I know, but the point here is that we should propagate `r`. If log_oom/_debug is used we synthesize `ENOMEM`... And we don't do that in library code